src/test/java/com/list/todo/services/TaskServiceTest.java

```diff
@@ -14,214 +12,233 @@
 
         // act
-        Optional<Task> addedTask = taskService.addTask(taskInput);
+        Optional<Task> addedTask = taskService.addTask(createTaskInput(TODO_LIST_ID));
 
         // assert
         assertEquals(addedTask, Optional.empty());
+        verify(todoListService).getTodoListById(TODO_LIST_ID);
     }
 
     @Test
-    public void updateTask_OnNonExistentTodoList_ReturnsAnObjectOfUpdatedTask() {
-
+    public void updateTask_OnExistentTask_ReturnsAnOptionalOfUpdatedTask() {
         // arrange
-        Long todoListId = 1L;
-
-        TodoList todoList = TodoList.builder()
-                .todoListName("todoList")
-                .tasks(new LinkedHashSet<>())
-                .build();
-        todoList.setId(todoListId);
-
-        Task oldTask = Task.builder()
-                .body("task")
-                .isComplete(false)
-                .todoList(todoList)
-                .build();
-
-        Long taskId = 2L;
-        oldTask.setId(taskId);
+        TaskInput taskInput = createTaskInput(TODO_LIST_ID);
+        TodoList todoList = createTodoList();
+        todoList.setId(TODO_LIST_ID);
+        Task oldTask = mock(Task.class);
+        oldTask.setId(TASK_ID_1);
         todoList.getTasks().add(oldTask);
+        Task updatedTask = createTask();
+        updatedTask.setId(TASK_ID_1);
 
-        Task updatedTask = Task.builder()
-                .body("updatedTask")
-                .isComplete(true)
-                .todoList(todoList)
-                .build();
-        updatedTask.setId(taskId);
-
-        when(taskRepositoryMock.findById(taskId)).thenReturn(Optional.of(oldTask));
-        when(taskRepositoryMock.save(oldTask)).thenReturn(updatedTask);
-
-        TaskInput taskInput = new TaskInput("task", false, 100L,  Priority.NOT_SPECIFIED, todoListId);
+        when(taskRepository.findById(TASK_ID_1)).thenReturn(Optional.of(oldTask));
+        when(taskRepository.save(oldTask)).thenReturn(updatedTask);
 
         // act
-        Optional<Task> taskFromService = taskService.updateTask(2L, taskInput);
+        Optional<Task> taskFromService = taskService.updateTask(TASK_ID_1, taskInput);
 
         // assert
-        assertEquals(taskFromService, Optional.of(updatedTask));
-
-        verify(taskRepositoryMock).save(any(Task.class));
+        assertEquals(updatedTask, taskFromService.get());
+        verify(taskRepository).findById(TASK_ID_1);
+        verify(taskRepository).save(oldTask);
+        verify(oldTask).setBody(taskInput.getBody());
+        verify(oldTask).setIsComplete(taskInput.getIsComplete());
+        verify(oldTask).setPriority(taskInput.getPriority());
+        verify(oldTask).setDurationTime(taskInput.getDurationTime());
+        verify(oldTask, times(0)).setCompletedDate(anyLong());
     }
 
     @Test
-    public void updateTask_updateNonExistentTask_ReturnsAnEmptyOptional() {
-
+    public void updateTask_SetIsCompleteToTrue_ReturnsAnOptionalOfUpdatedTask() {
```

Comment:
why would you need this test at all? isn't it same as updateTask_OnExistentTask_ReturnsAnOptionalOfUpdatedTask()